inspirehep/modules/workflows/tasks/refextract.py

```diff
@@ -191,42 +191,83 @@ def extract_references_from_raw_ref(reference, custom_kbs_file=None):
     )
 
 
-def match_reference(reference):
+def match_references(references):
+    """Match references to their respective records in INSPIRE.
+
+    Args:
+        references (list): the list of references
+
+    Retruns:
+        list: the matched references
+    """
+    matched_references, previous_matched_recid = [], None
+    for ref in references:
+        ref, previous_matched_recid = match_reference(ref, previous_matched_recid)
```

Comment:
you should only overwrite `previous_matched_recid` if it's not `None` (there might be additional references extracted that only contain garbage, and we don't want to consider those).